lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -58,6 +58,10 @@ def self.validate_MSFT_Credential(name, value)
         unless extraneous.empty?
           fail "#{name} includes invalid keys: #{extraneous.join(',')}"
         end
+
+        if value.values.any?{|v| v.nil? || v.length == 0}
+          fail "Both User and Password must not be empty"
```

Comment:
Initially thought we should allow an empty password as PSCredential allows it. However cannot create a SecureString with a null or empty string, so would fail there anyway.